lacci/lib/shoes/widgets/oval.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Shoes
+  class InvalidAttributeValueError < StandardError; end
+
+  class Oval < Shoes::Widget
+    include Shoes::Log
+    display_properties :left, :top, :width, :height, :center, :fill, :stroke, :strokewidth
+
+    def initialize(left = nil, top = nil, width = nil, height = nil, **options)
+      @left = convert_to_integer(left || options.fetch(:left), "left")
```

Comment:
Any particular reason left, top and width can't be negative, or that left and top can't be fractional?